static/app/components/globalFeedbackForm.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  createContext,
+  type ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react';
+import type {FeedbackDialog} from '@sentry/types';
+
+import {
+  type FeedbackIntegration,
+  useFeedback,
+  type UseFeedbackOptions,
+} from 'sentry/components/feedback/widget/useFeedback';
+
+interface GlobalFeedbackFormProps {
+  children: ReactNode;
+}
+
+interface GlobalFeedbackFormContextType {
+  feedback: FeedbackIntegration | undefined;
+  openForm: (options?: UseFeedbackOptions) => Promise<void>;
+}
+
+const GlobalFeedbackFormContext = createContext<GlobalFeedbackFormContextType>({
+  feedback: undefined,
+  openForm: Promise.resolve,
+});
+
+/**
+ * Returns two properties:
+ *
+ * - `feedback`: The feedback integration object. If undefined, the feedback
+ *               integration is not available and feedback buttons should not
+ *               be rendered.
+ * - `openForm`: A function that opens the feedback form. It accepts some
+ *               option overrides to change how the form is displayed.
+ *
+ *  Usage:
+ *
+ *  ```tsx
+ * const {feedback, openForm} = useFeedbackForm();
+ *
+ * if (!feedback) {
+ *   return null;
+ * }
+ *
+ * return <Button onClick={() => openForm()}>{t('Give Feedback')}</button>;
+ * ```
+ */
+export function useFeedbackForm() {
+  return useContext(GlobalFeedbackFormContext);
+}
+
+function useInternalFeedbackForm(props: UseFeedbackOptions) {
```

Comment:
I moved the old `useFeedbackForm` into this component so that there isn't any confusion on which one to use. I don't think it's helpful outside this component anymore